### PR TITLE
Use real DeepCopy instead of piecemeal implementation

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/kind feature
/area prow
/cc @cjwagner 
/assign @kargakis @BenTheElder

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
We want this as soon as we start touching more than just the env in the spec, manually copying things recursively is awful.

File: prow/pjutil/pjutil.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ package pjutil
 import (
 	"fmt"
 
-	uuid "github.com/satori/go.uuid"
+	"github.com/satori/go.uuid"
```
Comment (stevekuznetsov) [inline prow/pjutil/pjutil.go]: blame my editor and it's pesky cleanup?

**Comment by kargakis** [comment]
But I don't mind getting this in

File: prow/pjutil/pjutil.go [hunk 2]
```diff
@@ -144,23 +144,16 @@ func ProwJobToPod(pj kube.ProwJob, buildID string) (*v1.Pod, error) {
...
 	}
 
-	spec := pj.Spec.PodSpec
+	spec := pj.Spec.PodSpec.DeepCopy()
 	spec.RestartPolicy = "Never"
 
-	// Set environment variables in each container in the pod spec. We don't
-	// want to update the spec in place, since that will update the ProwJob
-	// spec. Instead, create a copy.
-	spec.InitContainers = []v1.Container{}
```
Comment (kargakis) [inline prow/pjutil/pjutil.go]: Won't this removal result in doubling the env variables inside the containers?

Comment (stevekuznetsov) [reply prow/pjutil/pjutil.go]: I don't think so? Before we started with an empty slice and copied each container over one by one, adding env to it as we went. Now we DeepCopy the slice and add env to each in place.

Comment (stevekuznetsov) [reply prow/pjutil/pjutil.go]: We can't DeepCopy _and_ do this assignment, or we copy the slice, then clear it with this assignment, then try to in-place edit an empty slice?

Comment (stevekuznetsov) [reply prow/pjutil/pjutil.go]: Also the unit tests would have caught a mismatch on env

**Pull request merged**